src/app/view/transport-call-creator/transport-call-creator.component.ts

```diff
@@ -205,9 +205,17 @@ export class TransportCallCreatorComponent implements OnInit {
   }
 
   showVesselPosition(): boolean {
-    if (!this.globals.config.enableVesselPositions) return false;
-    const selectedTimestamp = this.transportCallFormGroup.controls.timestampType.value;
-    return selectedTimestamp?.isVesselPositionNeeded ?? false;
+    let validators = null; 
+    const timestampTypeSelected = this.transportCallFormGroup.controls.timestampType.value;
+    if (timestampTypeSelected?.vesselPositionRequirement === EventLocationRequirement.REQUIRED 
+      && this.timestampChecking){
+      validators = [Validators.required]; 
+    }
+      this.transportCallFormGroup.controls.vesselPositionLatitude.setValidators(validators);
+      this.transportCallFormGroup.controls.vesselPositionLongitude.setValidators(validators);
+      this.transportCallFormGroup.controls.vesselPositionLatitude.updateValueAndValidity();
+      this.transportCallFormGroup.controls.vesselPositionLongitude.updateValueAndValidity();
+      return timestampTypeSelected?.vesselPositionRequirement !== EventLocationRequirement.EXCLUDED;
   }
```

Comment:
Ditto, weird indentation.

Reply:
Should probably have this done automatically heheh